service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/vm/CreateAzureVmStep.java

```diff
@@ -252,4 +243,63 @@ private VirtualMachine.DefinitionStages.WithCreate buildVmConfiguration(
     }
     return vmConfigurationFinalStep;
   }
+
+  private VirtualMachine.DefinitionStages.WithManagedCreate maybeAddEphemeralDiskStep(
+      VirtualMachine.DefinitionStages.WithManagedCreate priorSteps,
+      ApiAzureVmCreationParameters creationParameters) {
+    var maybePlacement =
+        Optional.ofNullable(creationParameters.getEphemeralOSDisk())
+            .flatMap(
+                (ephemeralOSDisk) -> {
+                  final Optional<DiffDiskPlacement> diskPlacement;
+                  switch (ephemeralOSDisk) {
```

Comment:
Could this be a place for Java 17 pattern matching? 
https://docs.oracle.com/en/java/javase/17/language/pattern-matching-switch-expressions-and-statements.html